.github/workflows/build.yml

```diff
@@ -8,44 +8,87 @@ on:
-    permissions:
-      id-token: write
-      attestations: write
     strategy:
       matrix:
         target_arch: [x86_64, x86, arm64-v8a, armeabi-v7a]
       fail-fast: false
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: Checkout repo
+        uses: actions/checkout@v4
         with:
           submodules: 'true'
 
-      - uses: nttld/setup-ndk@v1
+      - name: Setting up NDK
+        uses: nttld/setup-ndk@v1
         id: setup-ndk
         with:
-          ndk-version: r23c
+          ndk-version: r27c
           add-to-path: false
-      - run: ./build.sh ${{ matrix.target_arch }}
-        env:
-          NDK_TOOLCHAIN: ${{ steps.setup-ndk.outputs.ndk-path }}/toolchains/llvm/prebuilt/linux-x86_64
 
-      - name: Attest build provenance
-        uses: actions/attest-build-provenance@v1
-        with:
-          subject-name: dist-${{ matrix.target_arch }}
-          subject-path: dist
+      - name: Getting AOSP's source code
+        run: |
+          python3 ./get_source.py --tags platform-tools-35.0.2
+
+      - name: Compiling protoc
+        run: |
+          mkdir -p src/protobuf/build
+          pushd src/protobuf/build
+          cmake -GNinja -Dprotobuf_BUILD_TESTS=OFF ..
+          ninja -j$(nproc --all)
+          popd
+
+      - name: Compiling aapt2
+        run: python3 ./build.py --ndk=${{ steps.setup-ndk.outputs.ndk-path }} --abi=${{ matrix.target_arch }} --build="build/${{ matrix.target_arch }}" --protoc=/${{ github.workspace }}/src/protobuf/build/protoc --target aapt2
 
       - name: Upload artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: dist-${{ matrix.target_arch }}
-          path: dist
+          name: aapt2-${{ matrix.target_arch }}
+          path: ${{ github.workspace }}/build/${{ matrix.target_arch }}/bin/aapt2-*
+          if-no-files-found: error
+
+  release:
+    needs: build
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+      attestations: write
+    steps:
+      - name: Downloading artifact from all
+        uses: actions/download-artifact@v4
+        with:
+          path: ./artifacts
+
+      - name: List
+        run: |
+          ls -la ./artifacts/
+
+      - name: Creating release
+        uses: googleapis/release-please-action@v4
+        if : ${{ hashFiles('artifacts/*') != '' }}
+        id: release
+        with:
+          release-type: simple
+
+      - name: Generating attestation of the generated aapt2
+        if: ${{ steps.release.outputs.release_created }}
+        uses: actions/attest-build-provenance@v2
+        with:
+          subject-path: ./artifacts/aapt2-*/aapt2-*
+
+      - name: Uploading release artifact
+        if: ${{ steps.release.outputs.release_created }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I would like the workflow to follow the same design as the release workflow we have in other repos (such as the patches repo), so one release job, casing, naming and ideally similar steps, for consistency reasons. So that when we want a change we can apply them consistently across all repos.